pages/deployment/multicontainer.md

```diff
@@ -0,0 +1,80 @@
+---
+
+# Multicontainer Guide
+
+As your applications grow more complex, you may find significant benefit in running some services in separate containers. Multicontainer applications allow you to better isolate, duplicate, and manage key services, making your applications more modular and secure. Starting with resinOS vx.x.x, resin.io provides multicontainer application support, built around the [Docker Compose][docker-compose] tool. This guide will cover the considerations you need to take into account when using a multicontainer instance, including Docker Compose configuration, important resin.io specific settings, and single container applications.
+
+## Docker Compose
+
+At the root of your multicontainer application, you'll use a `docker-compose.yml` file to specify the configuration of your containers. The `docker-compose.yml` defines the services you'll be building, as well as how the services interact with each other and the host OS.
+
+Here's an example `docker-compose.yml` for a [simple web server][simple-web-server]:
+
+```
+ version: '2'
+ services:
+   web:
+     build: ./web
+     ports:
+      - "80:5000"
+     depends_on:
+      - redis
+   redis:
+    build: ./redis
+```
+
+Each service can either be built from a directory containing a Dockerfile, as shown here, or can use a **Docker** image that has already been built, by replacing `build:` with `image:`. If your containers need to started in a specific order, make sure to use the `depends_on:` setting.
+
+Unlike single container applications, multicontainer applications do not run containers in privileged mode by default. If you want to make use of hardware, you will either have to set some services to privileged, using `privileged: true`, or use the `cap_add` and `devices` settings to map in the correct hardware access to the container.
+
+Here, the `gpio` service is set up to use **i2c** sensors:
+
+```
+gpio:
+    build: ./gpio
+    devices:
+      - "/dev/i2c-1:/dev/i2c-1"
+      - "/dev/mem:/dev/mem"
+    cap_add: 
+      - SYS_RAWIO
+```
+
+## resin.io settings
+
+There are a few settings and considerations specific to resin.io that need to be taken into account when building multicontainer applications. First, using the `INITSYSTEM=on` setting in the Dockerfile of a service is only supported if the container is run as privileged, as **systemd** does not run correctly in unprivileged containers.
+
+In addition, there are some resin.io specific labels that can be defined in the `docker-compose.yml` file. These provide access to certain bind mounts and environment variables without requiring you to run the container as privileged.
+
+Label | Default | Description
+--- | --- | ---
+io.resin.features.dbus | false | Bind mounts the hostOS dbus into the container using “/run/dbus:/host/run/dbus”
+io.resin.features.kernel_modules | false | Bind mounts the hostOS /lib/modules into the container. (i.e. “/lib/modules:/lib/modules”)
```

Comment:
I believe all the feature labels here should have `-` rather than `_` in them, so `io.resin.features.kernel_modules` --> `io.resin.features.kernel-modules` as `_` are unsupported by docker